src/filesystem/impls/appshell/node/CSharpWatcher.js

```diff
@@ -0,0 +1,129 @@
+/*
+ * This copies the implementation of VSCode
+ * https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/services/files/node/watcher/win32/csharpWatcherService.ts
```

Comment:
Please, do not use the link with blob/master but the one with the commit SHA.
Maybe add the changes you made?